src/nvim/autocmd.c

```diff
@@ -1964,6 +1964,49 @@ void auto_next_pat(AutoPatCmd *apc, int stop_at_last)
   }
 }
 
+static bool call_autocmd_callback(const AutoCmd *ac, const AutoPat *ap)
+{
+  bool ret = false;
+  Callback callback = ac->exec.callable.cb;
+  if (callback.type == kCallbackLua) {
+    Dictionary data = ARRAY_DICT_INIT;
+    PUT(data, "match", CSTR_TO_OBJ((char *)autocmd_match));
+    PUT(data, "file", CSTR_TO_OBJ((char *)autocmd_fname));
+    PUT(data, "buf", INTEGER_OBJ(autocmd_bufnr));
+
+    Dictionary d = ARRAY_DICT_INIT;
+    PUT(d, "id", INTEGER_OBJ(ac->id));
+    PUT(d, "data", DICTIONARY_OBJ(data));
+    switch (ap->group) {
+    case AUGROUP_ERROR:
+      abort();  // unreachable
+    case AUGROUP_DEFAULT:
+    case AUGROUP_ALL:
+    case AUGROUP_DELETED:
+      PUT(d, "group", CSTR_TO_OBJ("NONE"));
```

Comment:
This means group will be either a string or an integer. Wouldn't it be better to just use `0` or `nil` here? `nil` would allow code like: 

```lua
if group then
  ... do something with group
end
```